pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -708,6 +741,123 @@ func initInfraMachineTemplateAndInfraClusterFromProvider(platform configv1.Platf
 	}
 }
 
+func (r *MachineSetSyncReconciler) reconcileMAPItoCAPIMachineSetDeletion(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (bool, error) {
+	logger := log.FromContext(ctx)
+
+	if mapiMachineSet.DeletionTimestamp.IsZero() {
+		if capiMachineSet == nil || capiMachineSet.DeletionTimestamp.IsZero() {
+			// Neither MAPI authoritative machine set nor its CAPI non-authoritative mirror
+			// are being deleted, nothing to reconcile for deletion.
+			return false, nil
+		}
+
+		// The MAPI authoritative machine set is not being deleted, but the CAPI non-authoritative one is.
```

Comment:
This is a little surprising to me. Is this a new edge case that was found during implementation?

I double checked the enhancement, I didn't see mention of deleting a mirror resource causing the authoritative one to be deleted.

Reply:
```Machine API 	- 	Delete 	Yes 	No 	Cluster API equivalent Machine remains in cluster```

says that the CAPI machine should stay.

```Cluster API 	- 	Delete 	Yes 	No 	Machine API synchronisation controller will recreate Machine after it is removed```

says that the MAPI machine should be re-created.

I see you've mentioned this in another comment, so I think my concern's being address in any case :)